### ISSUE TITLE
Move Working with the Code from CONTRIBUTING.adoc to the wiki

### ISSUE THREAD

**Issue by scottfrederick** [spring-projects/spring-boot#26332]
We'd like to move the [`Working with the Code` section](https://github.com/spring-projects/spring-boot/blob/main/CONTRIBUTING.adoc#working-with-the-code) from `CONTRIBUTING.adoc` to a new page on the wiki to make it easier to maintain.

**Comment by dreis2211** [comment]
If I may add my two cents here: I - as a contributor - always like having the contribution guides and everything that comes along with it inside the actual repository. Having that stuff in the Wiki always feels a bit decoupled and I personally barely look for instructions in the Wiki of a project. That might be wrong, but I wanted to give you the impression of someone that contributes from an outside perspective.

Moreover the Wiki has no versioning context by default, if I'm not mistaken. The switch from Maven to Gradle for example in the past was directly reflected in the specific markdown files inside the repo, which I personally see as a big benefit. I don't think we have that at the moment, but it's something that came to my mind.

I might be biased and since I'm no first-time-contributor anymore my view on this might be distorted, but I'd appreciate a link to the new Wiki section in `CONTRIBUTING.adoc` as a first timer.

Cheers,
Christoph

**Comment by scottfrederick** [comment]
Thanks for the feedback Christoph. As a frequent contributor this input is definitely valued. We started with a desire to put some of the more specific items like the existing `Cloning the git repository on Windows` and the suggestions in https://github.com/spring-projects/spring-boot/issues/25814#issuecomment-808742443 and https://github.com/spring-projects/spring-boot/issues/26309 on a wiki page to make them easier and faster to document, but ended up thinking it would be better to keep the "Working with the Code" content in one place. 

> Moreover the Wiki has no versioning context by default, if I'm not mistaken. The switch from Maven to Gradle for example in the past was directly reflected in the specific markdown files inside the repo, which I personally see as a big benefit. 

We discussed the fact that some of this info might vary by release branch, but that seemed manageable in the wiki format since build changes typically apply to a range of release branches. We'll keep an eye on that going forward. 

> I'd appreciate a link to the new Wiki section in CONTRIBUTING.adoc as a first timer.

We planned to link in both directions - from `CONTRIBUTING.adoc` to the wiki page and from the wiki page to `CONTRIBUTING.adoc`.
